app/poc-common/domain/mappers/createEncounterMapper.js

```diff
@@ -144,7 +145,43 @@ Poc.Common.CreateEncounterRequestMapper = (function () {
     };
 
     function isAnyObject(value) {
-        return value != null && typeof value === 'object';
+        var isObj;
+
+        if(value !== null && value !== undefined && typeof value === 'object') {
+            return true;
+        } else if (typeof value === 'string') {
+            try {
+                isObj = JSON.parse(value);
+
+                if(typeof isObj === 'object') {
+                    return true;
+                } else {
+                    return false;
+                }
+            } catch(err) {
+                return false;
+            }
+        }
+    }
+
+    function ensureObject(value) {
+        var isObj;
+
+        if(typeof value === 'object') {
+            return value;
+        } else if (typeof value === 'string') {
+            try {
+                isObj = JSON.parse(value);
+
+                if(typeof isObj === 'object') {
+                    return isObj;
+                } else {
+                    return null;
+                }
+            } catch(err) {
+                return null;
+            }
+        }
     }
```

Comment:
@steliomo and @valvijo will probably need to test that the functionality introduced above does not break their code. If there are some tests written in the `patient-schedule` branch, then we can wait for that branch to be merged and TravisCI will check this for us.